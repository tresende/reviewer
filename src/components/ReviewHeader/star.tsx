import { FaRegStar, FaStar } from 'react-icons/fa'
import theme from 'styles/theme'

export type StarProps = {
  outline?: boolean
}

const Star = ({ outline }: StarProps) => {
  const props = { color: theme.colors.yellow, size: 28 }
  return outline ? <FaRegStar {...props} /> : <FaStar {...props} />
}

export default Star
