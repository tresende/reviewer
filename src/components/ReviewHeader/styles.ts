import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.gray500};
    font-size: ${theme.font.sizes.xxlarge};
    a {
      color: ${theme.colors.primary};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
      &::before,
      &::after {
        content: ' ';
      }
    }
  `}
`

export const Location = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.normal};
    ${media.lessThan('small')`
        display: block;
      `}
  `}
`

export const Divider = styled.hr`
  ${({ theme }) => css`
    border: 0;
    margin: 1rem 0;
    border-top: ${theme.border.default} ${theme.colors.gray100};
  `}
`

export const Row = styled.div`
  display: flex;
  justify-content: space-between;
  ${media.lessThan('medium')`
    flex-direction: column;
  `}
`
export const IconWrapper = styled.div`
  display: flex;
  align-items: center;
  flex: 4;
  margin-bottom: 1rem;
`
export const NumReview = styled.span`
  ${({ theme }) => css`
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.medium};
  `}
`

export const ButtonWrapper = styled.div`
  display: flex;
  flex: 2;
`
