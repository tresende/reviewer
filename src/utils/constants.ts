const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MIN_TEXT_LENGTH = 5
const MAX_RATING = 5
const DEFAULT_SCORE = 3
const ANIMATION_DURATION = 1000

export { SCORE_LABEL, MAX_RATING, ANIMATION_DURATION, MIN_TEXT_LENGTH, DEFAULT_SCORE }
