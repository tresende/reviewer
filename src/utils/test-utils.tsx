import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
