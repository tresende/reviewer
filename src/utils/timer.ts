export function timeout(ms?: number | undefined) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}
