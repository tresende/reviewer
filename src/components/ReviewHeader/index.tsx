import * as S from './styles'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faStar } from '@fortawesome/free-solid-svg-icons'

const ReviewHeader = () => {
  library.add(faStar)
  return (
    <S.Wrapper>
      <S.Title>
        Avaliações de
        <a href="/coworking/space/bsb--brasilia">BSB Coworking</a>
        <span>Brasília/DF</span>
      </S.Title>
      <S.Divider />
      <div className="row">
        <div className="stars">
          <span className="icons-star">
            <FontAwesomeIcon size="1x" icon="star" />
          </span>
          <span className="num-review">5 avaliações</span>
        </div>
        <input
          id="writeReviewBtn"
          className="btn btn-block btn-outline btn-review pull-right"
          value="Avaliar"
          type="button"
        />
      </div>
    </S.Wrapper>
  )
}

export default ReviewHeader
