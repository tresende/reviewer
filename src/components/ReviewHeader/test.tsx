import { render, screen } from 'utils/test-utils'
import ReviewHeader from '.'

jest.mock('components/Stars', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="Stars" />
  }
}))

describe('<ReviewHeader />', () => {
  it('should render component correctly', () => {
    const onReview = jest.fn()
    const { container } = render(<ReviewHeader company="BORC" location="BH" onReview={onReview} />)

    expect(screen.getByText('BORC')).toBeInTheDocument()
    expect(screen.getByText('BH')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should execute callback when click in review', () => {
    const onReview = jest.fn()
    render(<ReviewHeader company="BORC" location="BH" onReview={onReview} />)

    screen.getByRole('button').click()
    expect(onReview).toBeCalledTimes(1)
  })
})
