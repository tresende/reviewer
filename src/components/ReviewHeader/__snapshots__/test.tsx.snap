// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewHeader /> should render component correctly 1`] = `
.c0 {
  color: #4b4b4b;
  font-size: 2.8rem;
}

.c1 {
  font-size: 1.8rem;
  font-weight: 400;
}

@media (max-width:450px) {
  .c1 {
    display: block;
  }
}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

<h1
  class="c0"
>
  Avaliações de
  <a
    href="#"
  >
     
    BORC
     
  </a>
  <span
    class="c1"
  >
    BH
  </span>
</h1>
`;
